### ISSUE TITLE
wrong number of arguments(0 fro 1+) appeared only on some pc

### ISSUE THREAD

**Issue by if1else0** [calabash/calabash-android#721]
error as below:
![1](https://cloud.githubusercontent.com/assets/10397913/15210693/2360f242-186a-11e6-9e9c-91114ae4e9e0.jpg)

when we debug it ,found that current android device serial was not in ports array in operations.rb file,but in other pc it does, error as below:

![image](https://cloud.githubusercontent.com/assets/10397913/15211989/f6b4ea9e-1870-11e6-990c-68afb83b4abb.png)

**Comment by TobiasRoikjer** [comment]
The error with tap appears because you are including `tap` from calabash into main scope. For example by having "include Calabash::Android::Operations" somewhere.

As for the device appearing, it is probably because you either don't have the right drivers on the computer, or your adb_usb.ini file has not been updated. To make the device appear on adb is out of the scope of Calabash-Android, but there are many resources online to help you with this

**Comment by if1else0** [comment]
Thanks for your quick reply. Actually, when i run adb devices, it can return the device serial, but it still has problem as picture 2 above.  And i did include Calabash::Android::Operations, but it can work in some PC,and some not,so i am so confused.

**Comment by if1else0** [comment]
And one more comment, i run the script on Mac OS X system. So i think the driver is not the problem.

**Comment by if1else0** [comment]
Thanks for your reply.It work now after I remove "include Calabash::Android::Operations". But if i remove the include sentence, how can i use the methods under the Operations module, thanks in advance.

**Comment by if1else0** [comment]
One more comment, in some PC, even i include "Calabash::Android::Operations", it still worked. I am so confused

**Comment by TobiasRoikjer** [comment]
If "works" because no other method happen to call the "tap" that is defined in the Ruby language.

"tap" is a kernel method, that is it is in the method lookup tree for every object in every scope of Ruby. When you include a module that has a new method tap, you add that module to the method lookup tree for the scope. The new tap has precedence over the kernel tap.

The Calabash tap method takes more than zero parameters (this is known as the methods arity). Say another library makes use of a tap method, they expect it to be the tap method of the Ruby language, not the one defined by Calabash. You can see the library "psych" is failing in your stack trace.

Long story short: Don't include Calabash::Android::Operations into main scope. Use World(module) for your cucumber steps.  Also notice that includes are cascading. If module A includes Calabash::Android::Operations, and another scope includes module A, then it implicitly also adds all Calabash methods to the lookup tree.
